vms/platformvm/txs/mempool/mempool_test.go

```diff
@@ -0,0 +1,872 @@
+				},
+			},
+			conflictsToRemove: []avax.UTXOID{
+				{TxID: ids.ID{123}},
+			},
+			wantTxs: []ids.ID{{1}, {2}},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			require := require.New(t)
+
+			m, err := New(
+				"",
+				gas.Dimensions{1, 1, 1, 1},
+				1_000_000,
+				snowtest.AVAXAssetID,
+				prometheus.NewRegistry(),
+			)
+			require.NoError(err)
+
+			for _, tx := range tt.txs {
+				require.NoError(m.Add(tx))
+			}
+
+			conflictsToRemove := set.Set[ids.ID]{}
+			for _, conflict := range tt.conflictsToRemove {
+				conflictsToRemove.Add(conflict.InputID())
+			}
+
+			m.RemoveConflicts(conflictsToRemove)
+
+			require.Equal(len(tt.wantTxs), m.Len())
+			for _, wantTxID := range tt.wantTxs {
+				_, ok := m.Get(wantTxID)
+				require.True(ok)
+			}
+		})
+	}
+}
+
+func TestMempool_Drop(t *testing.T) {
+	errFoo := errors.New("foo")
+
+	tests := []struct {
+		name       string
+		droppedTxs map[ids.ID]error
+		addTxs     []ids.ID
+		tx         ids.ID
+		wantErr    error
+	}{
+		{
+			name: "tx is dropped",
+			droppedTxs: map[ids.ID]error{
+				ids.Empty: errFoo,
+			},
+			tx:      ids.Empty,
+			wantErr: errFoo,
+		},
+		{
+			name: "tx dropped then added",
+			droppedTxs: map[ids.ID]error{
+				ids.Empty: errFoo,
+			},
+			addTxs: []ids.ID{ids.Empty},
+			tx:     ids.Empty,
+		},
+		{
+			name: "tx not dropped",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			require := require.New(t)
+
+			m, err := New(
+				"",
+				gas.Dimensions{1, 1, 1, 1},
+				1_000_000,
+				snowtest.AVAXAssetID,
+				prometheus.NewRegistry(),
+			)
+			require.NoError(err)
+
+			for txID, dropReason := range tt.droppedTxs {
+				m.MarkDropped(txID, dropReason)
+			}
+
+			for _, txID := range tt.addTxs {
+				require.NoError(m.Add(newTx(txID)))
+			}
+
+			err = m.GetDropReason(tt.tx)
+			require.ErrorIs(err, tt.wantErr)
+		})
+	}
+}
+
+func TestMempool_WaitForEvent(t *testing.T) {
+	require := require.New(t)
+
+	m, err := New(
+		"",
+		gas.Dimensions{1, 1, 1, 1},
+		1_000_000,
+		snowtest.AVAXAssetID,
+		prometheus.NewRegistry(),
+	)
+	require.NoError(err)
+
+	eg := &errgroup.Group{}
+	eg.Go(func() error {
+		tx := newTxWithUTXOs(
+			ids.GenerateTestID(),
+			[]*avax.TransferableInput{newAVAXInput(ids.GenerateTestID(), 5)},
+			1,
+		)
+
+		return m.Add(tx)
+	})
+
+	gotMsg, err := m.WaitForEvent(t.Context())
+	require.NoError(err)
+	require.Equal(common.PendingTxs, gotMsg)
+
+	require.NoError(eg.Wait())
+}
+
+func TestMempool_Iterate(t *testing.T) {
+	tests := []struct {
+		name    string
+		txs     []*txs.Tx
+		wantTxs []ids.ID
+	}{
+		{
+			name: "no txs",
+		},
+		{
+			name: "one tx",
+			txs: []*txs.Tx{
+				newTx(ids.ID{1}),
+			},
+			wantTxs: []ids.ID{
+				{1},
+			},
+		},
+		{
+			name: "multiple txs",
+			txs: []*txs.Tx{
+				newTx(ids.ID{1}),
+				newTx(ids.ID{2}),
+				newTx(ids.ID{3}),
+			},
+			wantTxs: []ids.ID{
+				{3},
+				{2},
+				{1},
+			},
+		},
```

Comment:
nit: can we lock in that we iterate in descending order here?

Reply:
Added some utxos to this